deep_qa/training/text_trainer.py

```diff
@@ -168,6 +168,39 @@ def create_data_arrays(self, dataset: IndexedDataset):
             return dataset.as_training_data()
 
     @overrides
+    def load_dataset_from_files(self, files: List[str]):
+        """
+        This method assumes you have a TextDataset that can be read from a single file.  If you
+        have something more complicated, you'll need to override this method (though, a solver that
+        has background information could call this method, then do additional processing on the
+        rest of the list, for instance).
+        """
+        return TextDataset.read_from_file(files[0], self._instance_type())
+
+    @overrides
+    def score_dataset(self, dataset: TextDataset):
+        """
+        See the superclass docs (:func:`Trainer.score_dataset`) for usage info.  Just a note here
+        that we `do not` use data generators for this method, even if you've said elsewhere that
+        you want to use them, so that we can easily return the labels for the data.  This means
+        that we'll do whole-dataset padding, and this could be slow.  We could probably fix this,
+        but it's good enough for now.
+        """
+        # TODO(matt): for some reason the reference to the super class docs above isn't getting
+        # linked properly.  I'm guessing it's because of an indexing issue in sphinx, but I
+        # couldn't figure it out.  Once that works, it can be changed to "See :func:`the superclass
+        # docs <Trainer.score_dataset>` for usage info").
+        # TODO(matt): it _should_ be as easy as iterating over the data again to pull out the
+        # labels, so we can still use data generators, but I'm waiting on implementing that.
+        indexed_dataset = dataset.to_indexed_dataset(self.data_indexer)
+        data_generator = self.data_generator
+        self.data_generator = None
```

Comment:
Add a comment here saying that this links up to the bit where you say we aren't using data generators, as it's not super obvious this is what it does atm.